vulkaninfo: New flags report warnings

The CMakeBuildList.txt is being updated to report more warnings and to report warnings as errors.
The following issues were found in vulkaninfo:
```
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(88): error C2220: warning treated as error - no 'object' file generated [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(88): warning C4456: declaration of 'arr' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(75): note: see declaration of 'arr'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(119): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(109): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(262): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(257): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(265): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(262): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(282): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(262): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(285): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(282): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(369): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(352): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(422): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(419): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(426): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(422): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(437): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(419): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(441): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(437): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(453): warning C4245: 'argument': conversion from 'int' to 'size_t', signed/unsigned mismatch [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(500): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(496): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(508): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(496): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(596): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(577): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(597): warning C4456: declaration of 'indent' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(578): note: see declaration of 'indent'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(659): warning C4456: declaration of 'obj' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(651): note: see declaration of 'obj'
D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(748): warning C4456: declaration of 'props' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\vulkaninfo\vulkaninfo.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\vulkaninfo\vulkaninfo.cpp(734): note: see declaration of 'props'
```

--- a/scripts/vulkaninfo_generator.py
+++ b/scripts/vulkaninfo_generator.py
@@ -441,7 +441,7 @@ def PrintGetFlagStrings(name, bitmask):
         "GetStrings(" + name + " value) {\n"
 
     out += "    std::vector<const char *> strings;\n"
-    out += "    if (value == 0) strings.push_back(\"None\");\n"
+    out += "    if (value == 0) { strings.push_back(\"None\"); return strings; }\n"
     for v in bitmask.options:
         val = v.value if isinstance(v.value, str) else str(hex(v.value))
         out += "    if (" + val + " & value) strings.push_back(\"" + \
@@ -454,13 +454,13 @@ def PrintFlags(bitmask, name):
     out = "void Dump" + name + \
         "(Printer &p, std::string name, " + name + " value, int width = 0) {\n"
     out += "    if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }\n"
-    out += "    auto strings = " + bitmask.name + \
-        "GetStrings(static_cast<" + bitmask.name + ">(value));\n"
     out += "    if (static_cast<" + bitmask.name + ">(value) == 0) {\n"
     out += "        ArrayWrapper arr(p, name, 0);\n"
     out += "        p.SetAsType().PrintString(\"None\");\n"
     out += "        return;\n"
     out += "    }\n"
+    out += "    auto strings = " + bitmask.name + \
+        "GetStrings(static_cast<" + bitmask.name + ">(value));\n"
     out += "    ArrayWrapper arr(p, name, strings.size());\n"
     out += "    for(auto& str : strings){\n"
     out += "        p.SetAsType().PrintString(str);\n"

--- a/vulkaninfo/generated/vulkaninfo.hpp
+++ b/vulkaninfo/generated/vulkaninfo.hpp
@@ -495,7 +495,7 @@ void DumpVkShaderFloatControlsIndependence(Printer &p, std::string name, VkShade
 }
 std::vector<const char *>VkCompositeAlphaFlagBitsKHRGetStrings(VkCompositeAlphaFlagBitsKHR value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("COMPOSITE_ALPHA_OPAQUE_BIT_KHR");
     if (0x2 & value) strings.push_back("COMPOSITE_ALPHA_PRE_MULTIPLIED_BIT_KHR");
     if (0x4 & value) strings.push_back("COMPOSITE_ALPHA_POST_MULTIPLIED_BIT_KHR");
@@ -504,12 +504,12 @@ std::vector<const char *>VkCompositeAlphaFlagBitsKHRGetStrings(VkCompositeAlphaF
 }
 void DumpVkCompositeAlphaFlagsKHR(Printer &p, std::string name, VkCompositeAlphaFlagsKHR value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkCompositeAlphaFlagBitsKHRGetStrings(static_cast<VkCompositeAlphaFlagBitsKHR>(value));
     if (static_cast<VkCompositeAlphaFlagBitsKHR>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkCompositeAlphaFlagBitsKHRGetStrings(static_cast<VkCompositeAlphaFlagBitsKHR>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -522,7 +522,7 @@ void DumpVkCompositeAlphaFlagBitsKHR(Printer &p, std::string name, VkCompositeAl
 
 std::vector<const char *>VkDeviceGroupPresentModeFlagBitsKHRGetStrings(VkDeviceGroupPresentModeFlagBitsKHR value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("DEVICE_GROUP_PRESENT_MODE_LOCAL_BIT_KHR");
     if (0x2 & value) strings.push_back("DEVICE_GROUP_PRESENT_MODE_REMOTE_BIT_KHR");
     if (0x4 & value) strings.push_back("DEVICE_GROUP_PRESENT_MODE_SUM_BIT_KHR");
@@ -531,12 +531,12 @@ std::vector<const char *>VkDeviceGroupPresentModeFlagBitsKHRGetStrings(VkDeviceG
 }
 void DumpVkDeviceGroupPresentModeFlagsKHR(Printer &p, std::string name, VkDeviceGroupPresentModeFlagsKHR value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkDeviceGroupPresentModeFlagBitsKHRGetStrings(static_cast<VkDeviceGroupPresentModeFlagBitsKHR>(value));
     if (static_cast<VkDeviceGroupPresentModeFlagBitsKHR>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkDeviceGroupPresentModeFlagBitsKHRGetStrings(static_cast<VkDeviceGroupPresentModeFlagBitsKHR>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -549,7 +549,7 @@ void DumpVkDeviceGroupPresentModeFlagBitsKHR(Printer &p, std::string name, VkDev
 
 std::vector<const char *>VkFormatFeatureFlagBitsGetStrings(VkFormatFeatureFlagBits value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("FORMAT_FEATURE_SAMPLED_IMAGE_BIT");
     if (0x2 & value) strings.push_back("FORMAT_FEATURE_STORAGE_IMAGE_BIT");
     if (0x4 & value) strings.push_back("FORMAT_FEATURE_STORAGE_IMAGE_ATOMIC_BIT");
@@ -585,12 +585,12 @@ std::vector<const char *>VkFormatFeatureFlagBitsGetStrings(VkFormatFeatureFlagBi
 }
 void DumpVkFormatFeatureFlags(Printer &p, std::string name, VkFormatFeatureFlags value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkFormatFeatureFlagBitsGetStrings(static_cast<VkFormatFeatureFlagBits>(value));
     if (static_cast<VkFormatFeatureFlagBits>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkFormatFeatureFlagBitsGetStrings(static_cast<VkFormatFeatureFlagBits>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -603,7 +603,7 @@ void DumpVkFormatFeatureFlagBits(Printer &p, std::string name, VkFormatFeatureFl
 
 std::vector<const char *>VkImageUsageFlagBitsGetStrings(VkImageUsageFlagBits value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("IMAGE_USAGE_TRANSFER_SRC_BIT");
     if (0x2 & value) strings.push_back("IMAGE_USAGE_TRANSFER_DST_BIT");
     if (0x4 & value) strings.push_back("IMAGE_USAGE_SAMPLED_BIT");
@@ -624,12 +624,12 @@ std::vector<const char *>VkImageUsageFlagBitsGetStrings(VkImageUsageFlagBits val
 }
 void DumpVkImageUsageFlags(Printer &p, std::string name, VkImageUsageFlags value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkImageUsageFlagBitsGetStrings(static_cast<VkImageUsageFlagBits>(value));
     if (static_cast<VkImageUsageFlagBits>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkImageUsageFlagBitsGetStrings(static_cast<VkImageUsageFlagBits>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -642,19 +642,19 @@ void DumpVkImageUsageFlagBits(Printer &p, std::string name, VkImageUsageFlagBits
 
 std::vector<const char *>VkMemoryHeapFlagBitsGetStrings(VkMemoryHeapFlagBits value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("MEMORY_HEAP_DEVICE_LOCAL_BIT");
     if (0x2 & value) strings.push_back("MEMORY_HEAP_MULTI_INSTANCE_BIT");
     return strings;
 }
 void DumpVkMemoryHeapFlags(Printer &p, std::string name, VkMemoryHeapFlags value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkMemoryHeapFlagBitsGetStrings(static_cast<VkMemoryHeapFlagBits>(value));
     if (static_cast<VkMemoryHeapFlagBits>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkMemoryHeapFlagBitsGetStrings(static_cast<VkMemoryHeapFlagBits>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -667,7 +667,7 @@ void DumpVkMemoryHeapFlagBits(Printer &p, std::string name, VkMemoryHeapFlagBits
 
 std::vector<const char *>VkMemoryPropertyFlagBitsGetStrings(VkMemoryPropertyFlagBits value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("MEMORY_PROPERTY_DEVICE_LOCAL_BIT");
     if (0x2 & value) strings.push_back("MEMORY_PROPERTY_HOST_VISIBLE_BIT");
     if (0x4 & value) strings.push_back("MEMORY_PROPERTY_HOST_COHERENT_BIT");
@@ -680,12 +680,12 @@ std::vector<const char *>VkMemoryPropertyFlagBitsGetStrings(VkMemoryPropertyFlag
 }
 void DumpVkMemoryPropertyFlags(Printer &p, std::string name, VkMemoryPropertyFlags value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkMemoryPropertyFlagBitsGetStrings(static_cast<VkMemoryPropertyFlagBits>(value));
     if (static_cast<VkMemoryPropertyFlagBits>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkMemoryPropertyFlagBitsGetStrings(static_cast<VkMemoryPropertyFlagBits>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -731,7 +731,7 @@ std::string VkQueueFlagsString(VkQueueFlags value, int width = 0) {
 }
 std::vector<const char *>VkResolveModeFlagBitsGetStrings(VkResolveModeFlagBits value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0 & value) strings.push_back("RESOLVE_MODE_NONE");
     if (0x1 & value) strings.push_back("RESOLVE_MODE_SAMPLE_ZERO_BIT");
     if (0x2 & value) strings.push_back("RESOLVE_MODE_AVERAGE_BIT");
@@ -741,12 +741,12 @@ std::vector<const char *>VkResolveModeFlagBitsGetStrings(VkResolveModeFlagBits v
 }
 void DumpVkResolveModeFlags(Printer &p, std::string name, VkResolveModeFlags value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkResolveModeFlagBitsGetStrings(static_cast<VkResolveModeFlagBits>(value));
     if (static_cast<VkResolveModeFlagBits>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkResolveModeFlagBitsGetStrings(static_cast<VkResolveModeFlagBits>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -759,7 +759,7 @@ void DumpVkResolveModeFlagBits(Printer &p, std::string name, VkResolveModeFlagBi
 
 std::vector<const char *>VkSampleCountFlagBitsGetStrings(VkSampleCountFlagBits value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("SAMPLE_COUNT_1_BIT");
     if (0x2 & value) strings.push_back("SAMPLE_COUNT_2_BIT");
     if (0x4 & value) strings.push_back("SAMPLE_COUNT_4_BIT");
@@ -771,12 +771,12 @@ std::vector<const char *>VkSampleCountFlagBitsGetStrings(VkSampleCountFlagBits v
 }
 void DumpVkSampleCountFlags(Printer &p, std::string name, VkSampleCountFlags value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkSampleCountFlagBitsGetStrings(static_cast<VkSampleCountFlagBits>(value));
     if (static_cast<VkSampleCountFlagBits>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkSampleCountFlagBitsGetStrings(static_cast<VkSampleCountFlagBits>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -789,7 +789,7 @@ void DumpVkSampleCountFlagBits(Printer &p, std::string name, VkSampleCountFlagBi
 
 std::vector<const char *>VkShaderStageFlagBitsGetStrings(VkShaderStageFlagBits value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("SHADER_STAGE_VERTEX_BIT");
     if (0x2 & value) strings.push_back("SHADER_STAGE_TESSELLATION_CONTROL_BIT");
     if (0x4 & value) strings.push_back("SHADER_STAGE_TESSELLATION_EVALUATION_BIT");
@@ -810,12 +810,12 @@ std::vector<const char *>VkShaderStageFlagBitsGetStrings(VkShaderStageFlagBits v
 }
 void DumpVkShaderStageFlags(Printer &p, std::string name, VkShaderStageFlags value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkShaderStageFlagBitsGetStrings(static_cast<VkShaderStageFlagBits>(value));
     if (static_cast<VkShaderStageFlagBits>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkShaderStageFlagBitsGetStrings(static_cast<VkShaderStageFlagBits>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -828,7 +828,7 @@ void DumpVkShaderStageFlagBits(Printer &p, std::string name, VkShaderStageFlagBi
 
 std::vector<const char *>VkSubgroupFeatureFlagBitsGetStrings(VkSubgroupFeatureFlagBits value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("SUBGROUP_FEATURE_BASIC_BIT");
     if (0x2 & value) strings.push_back("SUBGROUP_FEATURE_VOTE_BIT");
     if (0x4 & value) strings.push_back("SUBGROUP_FEATURE_ARITHMETIC_BIT");
@@ -842,12 +842,12 @@ std::vector<const char *>VkSubgroupFeatureFlagBitsGetStrings(VkSubgroupFeatureFl
 }
 void DumpVkSubgroupFeatureFlags(Printer &p, std::string name, VkSubgroupFeatureFlags value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkSubgroupFeatureFlagBitsGetStrings(static_cast<VkSubgroupFeatureFlagBits>(value));
     if (static_cast<VkSubgroupFeatureFlagBits>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkSubgroupFeatureFlagBitsGetStrings(static_cast<VkSubgroupFeatureFlagBits>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -860,18 +860,18 @@ void DumpVkSubgroupFeatureFlagBits(Printer &p, std::string name, VkSubgroupFeatu
 
 std::vector<const char *>VkSurfaceCounterFlagBitsEXTGetStrings(VkSurfaceCounterFlagBitsEXT value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("SURFACE_COUNTER_VBLANK_BIT_EXT");
     return strings;
 }
 void DumpVkSurfaceCounterFlagsEXT(Printer &p, std::string name, VkSurfaceCounterFlagsEXT value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkSurfaceCounterFlagBitsEXTGetStrings(static_cast<VkSurfaceCounterFlagBitsEXT>(value));
     if (static_cast<VkSurfaceCounterFlagBitsEXT>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkSurfaceCounterFlagBitsEXTGetStrings(static_cast<VkSurfaceCounterFlagBitsEXT>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -884,7 +884,7 @@ void DumpVkSurfaceCounterFlagBitsEXT(Printer &p, std::string name, VkSurfaceCoun
 
 std::vector<const char *>VkSurfaceTransformFlagBitsKHRGetStrings(VkSurfaceTransformFlagBitsKHR value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("SURFACE_TRANSFORM_IDENTITY_BIT_KHR");
     if (0x2 & value) strings.push_back("SURFACE_TRANSFORM_ROTATE_90_BIT_KHR");
     if (0x4 & value) strings.push_back("SURFACE_TRANSFORM_ROTATE_180_BIT_KHR");
@@ -898,12 +898,12 @@ std::vector<const char *>VkSurfaceTransformFlagBitsKHRGetStrings(VkSurfaceTransf
 }
 void DumpVkSurfaceTransformFlagsKHR(Printer &p, std::string name, VkSurfaceTransformFlagsKHR value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkSurfaceTransformFlagBitsKHRGetStrings(static_cast<VkSurfaceTransformFlagBitsKHR>(value));
     if (static_cast<VkSurfaceTransformFlagBitsKHR>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkSurfaceTransformFlagBitsKHRGetStrings(static_cast<VkSurfaceTransformFlagBitsKHR>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);
@@ -916,7 +916,7 @@ void DumpVkSurfaceTransformFlagBitsKHR(Printer &p, std::string name, VkSurfaceTr
 
 std::vector<const char *>VkToolPurposeFlagBitsEXTGetStrings(VkToolPurposeFlagBitsEXT value) {
     std::vector<const char *> strings;
-    if (value == 0) strings.push_back("None");
+    if (value == 0) { strings.push_back("None"); return strings; }
     if (0x1 & value) strings.push_back("TOOL_PURPOSE_VALIDATION_BIT_EXT");
     if (0x2 & value) strings.push_back("TOOL_PURPOSE_PROFILING_BIT_EXT");
     if (0x4 & value) strings.push_back("TOOL_PURPOSE_TRACING_BIT_EXT");
@@ -928,12 +928,12 @@ std::vector<const char *>VkToolPurposeFlagBitsEXTGetStrings(VkToolPurposeFlagBit
 }
 void DumpVkToolPurposeFlagsEXT(Printer &p, std::string name, VkToolPurposeFlagsEXT value, int width = 0) {
     if (p.Type() == OutputType::json) { p.PrintKeyValue(name, value); return; }
-    auto strings = VkToolPurposeFlagBitsEXTGetStrings(static_cast<VkToolPurposeFlagBitsEXT>(value));
     if (static_cast<VkToolPurposeFlagBitsEXT>(value) == 0) {
         ArrayWrapper arr(p, name, 0);
         p.SetAsType().PrintString("None");
         return;
     }
+    auto strings = VkToolPurposeFlagBitsEXTGetStrings(static_cast<VkToolPurposeFlagBitsEXT>(value));
     ArrayWrapper arr(p, name, strings.size());
     for(auto& str : strings){
         p.SetAsType().PrintString(str);

--- a/vulkaninfo/outputprinter.h
+++ b/vulkaninfo/outputprinter.h
@@ -272,6 +272,16 @@ class Printer {
         return *this;
     }
 
+    Printer &SetIgnoreMinWidth() {
+        ignore_min_width_parameter = true;
+        return *this;
+    }
+
+    Printer &UnsetIgnoreMinWidth() {
+        ignore_min_width_parameter = false;
+        return *this;
+    }
+
     Printer &SetElementIndex(int index) {
         assert(index >= 0 && "cannot set element index to a negative value");
         element_index = index;
@@ -462,10 +472,9 @@ class Printer {
     void PrintKeyValue(std::string key, T value, size_t min_key_width = 0, std::string value_description = "") {
         switch (output_type) {
             case (OutputType::text):
-                if (min_key_width > key.size()) {
-                    out << std::string(static_cast<size_t>(indents), '\t') << key << std::string(min_key_width - key.size(), ' ');
-                } else {
-                    out << std::string(static_cast<size_t>(indents), '\t') << key;
+                out << std::string(static_cast<size_t>(indents), '\t') << key;
+                if (min_key_width > key.size() && !ignore_min_width_parameter) {
+                    out << std::string(min_key_width - key.size(), ' ');
                 }
                 out << " = " << value;
                 if (value_description != "") {
@@ -673,6 +682,8 @@ class Printer {
 
     // make object titles the color of types
     bool set_object_name_as_type = false;
+
+    bool ignore_min_width_parameter = false;
 
     // objects which are in an array
     int element_index = -1;  // negative one is the sentinel value

--- a/vulkaninfo/outputprinter.h
+++ b/vulkaninfo/outputprinter.h
@@ -388,12 +388,12 @@ class Printer {
                 break;
         }
     }
-    void ArrayStart(std::string array_name, int32_t element_count = 0) {
+    void ArrayStart(std::string array_name, size_t element_count = 0) {
         switch (output_type) {
             case (OutputType::text): {
                 out << std::string(static_cast<size_t>(indents), '\t') << array_name << ":";
                 size_t underline_count = array_name.size() + 1;
-                if (element_count >= 0) {
+                if (element_count > 0) {
                     out << " count = " << element_count;
                     underline_count += 9 + std::to_string(element_count).size();
                 }
@@ -410,7 +410,7 @@ class Printer {
                     out << "<details>";
                 }
                 out << "<summary>" << array_name;
-                if (element_count >= 0) {
+                if (element_count > 0) {
                     out << ": count = <span class='val'>" << element_count << "</span>";
                 }
                 out << "</summary>\n";
@@ -725,7 +725,7 @@ class ObjectWrapper {
 class ArrayWrapper {
   public:
     ArrayWrapper(Printer &p, std::string array_name, size_t element_count = 0) : p(p) {
-        p.ArrayStart(array_name, static_cast<int32_t>(element_count));
+        p.ArrayStart(array_name, element_count);
     }
     ~ArrayWrapper() { p.ArrayEnd(); }
 

--- a/vulkaninfo/vulkaninfo.cpp
+++ b/vulkaninfo/vulkaninfo.cpp
@@ -161,13 +161,7 @@ void DumpSurfaceCapabilities(Printer &p, AppInstance &inst, AppGpu &gpu, AppSurf
     if (inst.CheckExtensionEnabled(VK_EXT_DISPLAY_SURFACE_COUNTER_EXTENSION_NAME)) {
         p.SetSubHeader();
         ObjectWrapper obj(p, "VkSurfaceCapabilities2EXT");
-        {
-            ArrayWrapper arr(p, "supportedSurfaceCounters");
-            if (surface.surface_capabilities2_ext.supportedSurfaceCounters == 0) p.PrintString("None");
-            if (surface.surface_capabilities2_ext.supportedSurfaceCounters & VK_SURFACE_COUNTER_VBLANK_EXT) {
-                p.SetAsType().PrintString("VK_SURFACE_COUNTER_VBLANK_EXT");
-            }
-        }
+        DumpVkSurfaceCounterFlagsEXT(p, "supportedSurfaceCounters", surface.surface_capabilities2_ext.supportedSurfaceCounters);
     }
     if (inst.CheckExtensionEnabled(VK_KHR_GET_SURFACE_CAPABILITIES_2_EXTENSION_NAME)) {
         chain_iterator_surface_capabilities2(p, inst, gpu, surface.surface_capabilities2_khr.pNext, inst.vk_version);
@@ -450,7 +444,7 @@ void GpuDumpMemoryProps(Printer &p, AppGpu &gpu) {
             // only linear and optimal tiling considered
             for (auto &image_tiling : gpu.memory_image_support_types) {
                 p.SetOpenDetails();
-                ArrayWrapper arr(p, VkImageTilingString(VkImageTiling(image_tiling.tiling)), 0);
+                ArrayWrapper arr(p, VkImageTilingString(VkImageTiling(image_tiling.tiling)));
                 bool has_any_support_types = false;
                 bool regular = false;
                 bool transient = false;

--- a/vulkaninfo/vulkaninfo.cpp
+++ b/vulkaninfo/vulkaninfo.cpp
@@ -155,8 +155,9 @@ void DumpPresentModes(Printer &p, AppSurface &surface) {
 
 void DumpSurfaceCapabilities(Printer &p, AppInstance &inst, AppGpu &gpu, AppSurface &surface) {
     auto &surf_cap = surface.surface_capabilities;
-    p.SetSubHeader();
+    p.SetSubHeader().SetIgnoreMinWidth();
     DumpVkSurfaceCapabilitiesKHR(p, "VkSurfaceCapabilitiesKHR", surf_cap);
+    p.UnsetIgnoreMinWidth();
 
     if (inst.CheckExtensionEnabled(VK_EXT_DISPLAY_SURFACE_COUNTER_EXTENSION_NAME)) {
         p.SetSubHeader();

--- a/vulkaninfo/vulkaninfo.cpp
+++ b/vulkaninfo/vulkaninfo.cpp
@@ -72,7 +72,7 @@ void DumpLayers(Printer &p, std::vector<LayerExtensionList> layers, const std::v
         case OutputType::text:
         case OutputType::html: {
             p.SetHeader();
-            ArrayWrapper arr(p, "Layers", layers.size());
+            ArrayWrapper arr_layers(p, "Layers", layers.size());
             IndentWrapper indent(p);
 
             for (auto &layer : layers) {
@@ -85,7 +85,7 @@ void DumpLayers(Printer &p, std::vector<LayerExtensionList> layers, const std::v
                 ObjectWrapper obj(p, header);
                 DumpExtensions(p, "Layer", layer.extension_properties);
 
-                ArrayWrapper arr(p, "Devices", gpus.size());
+                ArrayWrapper arr_devices(p, "Devices", gpus.size());
                 for (auto &gpu : gpus) {
                     p.PrintKeyValue("GPU id", gpu->id, 0, gpu->props.deviceName);
                     auto exts = gpu->AppGetPhysicalDeviceLayerExtensions(props.layerName);
@@ -116,7 +116,7 @@ void DumpLayers(Printer &p, std::vector<LayerExtensionList> layers, const std::v
                 DumpExtensions(p, "Layer", layer.extension_properties);
                 ObjectWrapper obj_devices(p, "Devices");
                 for (auto &gpu : gpus) {
-                    ObjectWrapper obj(p, gpu->props.deviceName);
+                    ObjectWrapper obj_gpu(p, gpu->props.deviceName);
                     p.PrintKeyValue("GPU id", gpu->id, 0, gpu->props.deviceName);
                     auto exts = gpu->AppGetPhysicalDeviceLayerExtensions(layer.layer_properties.layerName);
                     DumpExtensions(p, "Layer-Device", exts);
@@ -254,15 +254,15 @@ void DumpGroups(Printer &p, AppInstance &inst) {
         }
 
         p.SetHeader();
-        ObjectWrapper obj(p, "Device Groups");
+        ObjectWrapper obj_device_groups(p, "Device Groups");
         IndentWrapper indent(p);
 
         int group_id = 0;
         for (auto &group : groups) {
-            ObjectWrapper obj(p, "Group " + std::to_string(group_id));
+            ObjectWrapper obj_group(p, "Group " + std::to_string(group_id));
             auto group_props = GetGroupProps(inst, group);
             {
-                ObjectWrapper obj(p, "Properties");
+                ObjectWrapper obj_properties(p, "Properties");
                 {
                     ArrayWrapper arr(p, "physicalDevices", group.physicalDeviceCount);
                     int id = 0;
@@ -279,9 +279,9 @@ void DumpGroups(Printer &p, AppInstance &inst) {
                 p.PrintKeyString("Present Capabilities",
                                  "Group does not support VK_KHR_device_group, skipping printing present capabilities");
             } else {
-                ObjectWrapper obj(p, "Present Capabilities");
+                ObjectWrapper obj_caps(p, "Present Capabilities");
                 for (uint32_t i = 0; i < group.physicalDeviceCount; i++) {
-                    ObjectWrapper obj(
+                    ObjectWrapper obj_device(
                         p, std::string(group_props[i].deviceName) + " (ID: " + p.DecorateAsValue(std::to_string(i)) + ")");
                     ArrayWrapper arr(p, "Can present images from the following devices", group.physicalDeviceCount);
 
@@ -349,7 +349,7 @@ void GpuDumpPropsJson(Printer &p, AppGpu &gpu) {
 void GpuDumpQueueProps(Printer &p, std::vector<SurfaceExtension> &surfaces, const AppQueueFamilyProperties &queue) {
     VkQueueFamilyProperties props = queue.props;
     p.SetSubHeader().SetElementIndex(static_cast<int>(queue.queue_index));
-    ObjectWrapper obj(p, "queueProperties");
+    ObjectWrapper obj_queue_props(p, "queueProperties");
     if (p.Type() == OutputType::vkconfig_output) {
         DumpVkExtent3D(p, "minImageTransferGranularity", props.minImageTransferGranularity);
     } else {
@@ -366,7 +366,7 @@ void GpuDumpQueueProps(Printer &p, std::vector<SurfaceExtension> &surfaces, cons
         for (auto &surface : surfaces) {
             if (surface.name.size() > width) width = surface.name.size();
         }
-        ObjectWrapper obj(p, "present support");
+        ObjectWrapper obj_present_support(p, "present support");
         for (auto &surface : surfaces) {
             p.PrintKeyString(surface.name, surface.supports_present ? "true" : "false", width);
         }
@@ -416,14 +416,14 @@ std::string append_human_readible(VkDeviceSize memory) {
 
 void GpuDumpMemoryProps(Printer &p, AppGpu &gpu) {
     p.SetHeader();
-    ObjectWrapper obj(p, "VkPhysicalDeviceMemoryProperties");
+    ObjectWrapper obj_mem_props(p, "VkPhysicalDeviceMemoryProperties");
     IndentWrapper indent(p);
     {
-        ObjectWrapper obj(p, "memoryHeaps", gpu.memory_props.memoryHeapCount);
+        ObjectWrapper obj_mem_heaps(p, "memoryHeaps", gpu.memory_props.memoryHeapCount);
 
         for (uint32_t i = 0; i < gpu.memory_props.memoryHeapCount; ++i) {
             p.SetElementIndex(static_cast<int>(i));
-            ObjectWrapper obj(p, "memoryHeaps");
+            ObjectWrapper obj_mem_heap(p, "memoryHeaps");
 
             p.PrintKeyString("size", append_human_readible(gpu.memory_props.memoryHeaps[i].size), 6);
             if (gpu.CheckPhysicalDeviceExtensionIncluded(VK_EXT_MEMORY_BUDGET_EXTENSION_NAME)) {
@@ -434,11 +434,11 @@ void GpuDumpMemoryProps(Printer &p, AppGpu &gpu) {
         }
     }
     {
-        ObjectWrapper obj(p, "memoryTypes", gpu.memory_props.memoryTypeCount);
+        ObjectWrapper obj_mem_types(p, "memoryTypes", gpu.memory_props.memoryTypeCount);
 
         for (uint32_t i = 0; i < gpu.memory_props.memoryTypeCount; ++i) {
             p.SetElementIndex(static_cast<int>(i));
-            ObjectWrapper obj(p, "memoryTypes");
+            ObjectWrapper obj_mem_type(p, "memoryTypes");
             p.PrintKeyValue("heapIndex", gpu.memory_props.memoryTypes[i].heapIndex, 13);
 
             auto flags = gpu.memory_props.memoryTypes[i].propertyFlags;
@@ -450,7 +450,7 @@ void GpuDumpMemoryProps(Printer &p, AppGpu &gpu) {
             // only linear and optimal tiling considered
             for (auto &image_tiling : gpu.memory_image_support_types) {
                 p.SetOpenDetails();
-                ArrayWrapper arr(p, VkImageTilingString(VkImageTiling(image_tiling.tiling)), -1);
+                ArrayWrapper arr(p, VkImageTilingString(VkImageTiling(image_tiling.tiling)), 0);
                 bool has_any_support_types = false;
                 bool regular = false;
                 bool transient = false;
@@ -493,7 +493,7 @@ void GpuDumpMemoryProps(Printer &p, AppGpu &gpu) {
 }
 
 void GpuDumpMemoryPropsJson(Printer &p, AppGpu &gpu) {
-    ObjectWrapper obj(p, "VkPhysicalDeviceMemoryProperties");
+    ObjectWrapper obj_mem_props(p, "VkPhysicalDeviceMemoryProperties");
     {
         ArrayWrapper arr(p, "memoryHeaps", gpu.memory_props.memoryHeapCount);
         for (uint32_t i = 0; i < gpu.memory_props.memoryHeapCount; ++i) {
@@ -574,8 +574,8 @@ void GpuDumpToolingInfo(Printer &p, AppGpu &gpu) {
 
 void GpuDevDump(Printer &p, AppGpu &gpu) {
     p.SetHeader();
-    ObjectWrapper obj(p, "Format Properties");
-    IndentWrapper indent(p);
+    ObjectWrapper obj_format_props(p, "Format Properties");
+    IndentWrapper indent_outer(p);
 
     if (p.Type() == OutputType::text) {
         auto fmtPropMap = FormatPropMap(gpu);
@@ -593,10 +593,10 @@ void GpuDevDump(Printer &p, AppGpu &gpu) {
             }
 
             p.SetElementIndex(counter++);
-            ObjectWrapper obj(p, "Common Format Group");
-            IndentWrapper indent(p);
+            ObjectWrapper obj_common_group(p, "Common Format Group");
+            IndentWrapper indent_inner(p);
             {
-                ArrayWrapper arr(p, "Formats", prop.second.size());
+                ArrayWrapper arr_formats(p, "Formats", prop.second.size());
                 for (auto &fmt : prop.second) {
                     p.SetAsType().PrintString(VkFormatString(fmt));
                 }
@@ -605,7 +605,7 @@ void GpuDevDump(Printer &p, AppGpu &gpu) {
             p.AddNewline();
         }
 
-        ArrayWrapper arr(p, "Unsupported Formats", unsupported_formats.size());
+        ArrayWrapper arr_unsupported_formats(p, "Unsupported Formats", unsupported_formats.size());
         for (auto &fmt : unsupported_formats) {
             p.SetAsType().PrintString(VkFormatString(fmt));
         }
@@ -648,7 +648,7 @@ void GpuDevDumpJson(Printer &p, AppGpu &gpu) {
 // Print gpu info for text, html, & vkconfig_output
 // Uses a seperate function than schema-json for clarity
 void DumpGpu(Printer &p, AppGpu &gpu, bool show_formats) {
-    ObjectWrapper obj(p, "GPU" + std::to_string(gpu.id));
+    ObjectWrapper obj_gpu(p, "GPU" + std::to_string(gpu.id));
     IndentWrapper indent(p);
 
     GpuDumpProps(p, gpu);
@@ -656,7 +656,7 @@ void DumpGpu(Printer &p, AppGpu &gpu, bool show_formats) {
     p.AddNewline();
     {
         p.SetHeader();
-        ObjectWrapper obj(p, "VkQueueFamilyProperties");
+        ObjectWrapper obj_family_props(p, "VkQueueFamilyProperties");
         for (const auto &queue_prop : gpu.extended_queue_props) {
             GpuDumpQueueProps(p, gpu.inst.surface_extensions, queue_prop);
         }
@@ -745,11 +745,11 @@ void DumpSummaryGPU(Printer &p, AppGpu &gpu) {
         while (place) {
             struct VkStructureHeader *structure = (struct VkStructureHeader *)place;
             if (structure->sType == VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DRIVER_PROPERTIES) {
-                VkPhysicalDeviceDriverProperties *props = (VkPhysicalDeviceDriverProperties *)structure;
-                DumpVkDriverId(p, "driverID", props->driverID, 18);
-                p.PrintKeyString("driverName", props->driverName, 18);
-                p.PrintKeyString("driverInfo", props->driverInfo, 18);
-                DumpVkConformanceVersion(p, "conformanceVersion", props->conformanceVersion, 18);
+                VkPhysicalDeviceDriverProperties *driver_props = (VkPhysicalDeviceDriverProperties *)structure;
+                DumpVkDriverId(p, "driverID", driver_props->driverID, 18);
+                p.PrintKeyString("driverName", driver_props->driverName, 18);
+                p.PrintKeyString("driverInfo", driver_props->driverInfo, 18);
+                DumpVkConformanceVersion(p, "conformanceVersion", driver_props->conformanceVersion, 18);
             }
             place = structure->pNext;
         }

